src/features/processPicker.ts

```diff
@@ -35,6 +37,125 @@ export class AttachPicker {
+        for (configIdx = 0; configIdx < json.configurations.length; ++configIdx) {
+            if (json.configurations[configIdx].name === name) {
+                config = json.configurations[configIdx];
+                break;
+            }
+        }
+
+        if (configIdx == json.configurations.length) {
+            // Name not found in list of given configurations. 
+            return Promise.reject<string>(new Error(name + " could not be found in configurations."));
+        }
+
+        if (!config.pipeTransport || !config.pipeTransport.debuggerPath) {
+            // Missing PipeTransport and debuggerPath, prompt if user wanted to just do local attach.
+            return Promise.reject<string>(new Error("Configuration \"" + name + "\" in launch.json does not have a " +
+                "pipeTransport argument with debuggerPath for pickRemoteProcess. Use pickProcess for local attach."));
+        } else {
+            let pipeProgram = config.pipeTransport.pipeProgram;
+            let pipeArgs = config.pipeTransport.pipeArgs;
+            let platformSpecificPipeTransportOptions = RemoteAttachPicker.getPlatformSpecificPipeTransportOptions(config);
+
+            if (platformSpecificPipeTransportOptions) {
+                pipeProgram = platformSpecificPipeTransportOptions.pipeProgram || pipeProgram;
+                pipeArgs = platformSpecificPipeTransportOptions.pipeArgs || pipeArgs;
+            }
+
+            let pipeCmd: string = pipeProgram + " " + pipeArgs.join(" ");
+            return RemoteAttachPicker.getRemoteOSAndProcesses(pipeCmd).then(processes => {
+                let attachPickOptions: vscode.QuickPickOptions = {
+                    matchOnDescription: true,
+                    matchOnDetail: true,
+                    placeHolder: "Select the process to attach to"
+                };
+                return vscode.window.showQuickPick(processes, attachPickOptions).then(item => {
+                    return item ? item.id : null;
+                });
+            });
+        }
+    }
+
+    private static getPlatformSpecificPipeTransportOptions(config) {
+        const osPlatform = os.platform();
+
+        if (osPlatform == "darwin" && config.pipeTransport.osx) {
+            return config.pipeTransport.osx;
+        } else if (osPlatform == "linux" && config.pipeTransport.linux) {
+            return config.pipeTransport.linux;
+        } else if (osPlatform == "win32" && config.pipeTransport.windows) {
+            return config.pipeTransport.windows;
+        }
+
+        return null;
+    }
+
+    public static getRemoteOSAndProcesses(pipeCmd: string): Promise<AttachItem[]> {
+        // Commands to get OS and processes
+        const command = `uname && if [ $(uname) == "Linux" ] ; then ${RemoteAttachPicker.linuxPsCommand} ; elif [ $(uname) == "Darwin" ] ; ` +
+            `then ${RemoteAttachPicker.osxPsCommand}; fi`;
+
+        return execChildProcess(`${pipeCmd} "${command}"`, null).then(output => {
```

Comment:
I added this comment earlier, but to a spot where it got lost -- currently we will just reject the promise if the command returns stderror text. This might be confusing now that we are using execChildProcess for the pipe program. I would suggest doing something like putting in the wrong ip address in your `plink` or `ssh` command line and see if the experience is compressible.